data/crac/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation_util/UcteConnectable.java

```diff
@@ -0,0 +1,105 @@
+        BOTH // used for all elements but tie-lines
+    }
+
+    UcteConnectable(String from, String to, String orderCode, Set<String> elementNames, Identifiable<?> iidmConnectable, boolean isIidmConventionInverted) {
+        this(from, to, orderCode, elementNames, iidmConnectable, isIidmConventionInverted, Side.BOTH);
+    }
+
+    UcteConnectable(String from, String to, String orderCode, Set<String> elementNames, Identifiable<?> iidmConnectable, boolean isIidmConventionInverted, Side side) {
+        this.ucteFromNode = from;
+        this.ucteToNode = to;
+        if (this.ucteFromNode.length() != UCTE_NODE_LENGTH || this.ucteToNode.length() != UCTE_NODE_LENGTH) {
+            throw new IllegalArgumentException(String.format("from (%s) and to (%s) should have %d characters", this.ucteFromNode, this.ucteToNode, UCTE_NODE_LENGTH));
+        }
+        this.ucteOrderCode = orderCode;
+        this.ucteElementNames = elementNames;
+        this.iidmIdentifiable = iidmConnectable;
+        this.isIidmConventionInverted = isIidmConventionInverted;
+        this.iidmSide = side;
+    }
+
+    boolean doesMatch(String from, String to, String suffix) {
+        return matchSuffix(suffix) && matchFromTo(from, to);
+    }
+
+    UcteMatchingResult getUcteMatchingResult(String from, String to, String suffix) {
+        if (!matchSuffix(suffix) || !matchFromTo(from, to)) {
+            return UcteMatchingResult.notFound();
+        } else {
+            return UcteMatchingResult.found(iidmSide, isIidmConventionInverted, iidmIdentifiable);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%1$-8s %2$-8s %3$s - %4$s - side %5$s", ucteFromNode, ucteToNode, ucteOrderCode, iidmIdentifiable.getId(), iidmSide);
+    }
+
+    @Override
+    public int hashCode() {
+        return toString().hashCode();
+    }
+
+    @Override
+    public int compareTo(UcteConnectable o) {
+        return this.hashCode() - o.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof UcteConnectable) {
+            UcteConnectable other = (UcteConnectable) obj;
+            return this.ucteFromNode.equals(other.ucteFromNode)
+                && this.ucteToNode.equals(other.ucteToNode)
+                && this.ucteOrderCode.equals(other.ucteOrderCode)
+                && this.ucteElementNames.equals(other.ucteElementNames)
+                && this.iidmIdentifiable.getId().equals(other.iidmIdentifiable.getId())
+                && this.iidmSide.equals(other.iidmSide);
+        }
+        return false;
+    }
+
+    private boolean matchFromTo(String from, String to) {
+        return UcteUtils.matchNodeNames(from, this.ucteFromNode) && UcteUtils.matchNodeNames(to, this.ucteToNode);
+    }
+
+    private boolean matchSuffix(String suffix) {
+        return (suffix.equals(ucteOrderCode)) || (ucteElementNames != null && ucteElementNames.contains(suffix));
```

Comment:
We observed something of the kind last year with @TheMaskedTurtle.

Two lines with the same 7th caracthers in the from/to node names, one with order code "1", the other one with elementName "1" (and order code != "1").

This case was put aside as we want the 8 characters of the from/to node names to match, instead of the 7th first ones, in the CORE CC. Moreover, a line with elementName = "1" and orderCode != "1" was more looking like a data mistake, than something we could reasonably expect.

However, if we want something more robust we could indeed be more specific on the suffix, being clear on whether it is an orderCode or an elementName.